### PR TITLE
AddPeer in one consensus operation

### PR DESCRIPTION
**Pull request by e-ivkov**
Adds a peer similar to the removal for consistency. PR tries to stay compatible with the previous version so that migration can happen without downtime.

Closes https://github.com/qdrant/qdrant/issues/915

### CODE REVIEW

File: lib/storage/src/content_manager/consensus_state.rs [hunk 1]
```diff
@@ -227,6 +263,7 @@ impl<C: CollectionContainer> ConsensusState<C> {
             };
             let do_increase_applied_index: bool = if entry.data.is_empty() {
                 // Empty entry, when the peer becomes Leader it will send an empty entry.
+                self.is_leader_established.make_ready();
```
Comment (agourlay) [inline lib/storage/src/content_manager/consensus_state.rs:266]: Why is this a better location to set `make_ready()`?

Comment (e-ivkov) [reply lib/storage/src/content_manager/consensus_state.rs:266]: I tried to remove all sleeps from the consensus test. There we assert that we have this empty entry when the leader was elected, and as you can notice it happens a few lines after the conf change.

So this just gives us stricter guarantees.

File: lib/storage/src/content_manager/consensus_state.rs [hunk 2]
```diff
@@ -173,17 +171,55 @@ impl<C: CollectionContainer> ConsensusState<C> {
...
-                log::debug!("Remove node {}", single_change.node_id);
-                if self.this_peer_id() == single_change.node_id {
-                    stop_consensus = true;
+            match single_change.change_type() {
+                ConfChangeType::AddNode => {
+                    debug_assert!(
+                        self.peer_address_by_id()
+                            .get(&single_change.node_id)
+                            .is_some(),
+                        "Peer should be already known"
```
Comment (generall) [inline lib/storage/src/content_manager/consensus_state.rs:180]: is that because we always use `AddLearnerNode` first?

Comment (e-ivkov) [reply lib/storage/src/content_manager/consensus_state.rs:180]: Yes

**Review by generall — APPROVED** [review]

**Pull request merged**